File: code/ex392.swift
class Solution {
    func isSubsequence(_ s: String, _ t: String) -> Bool {
        let arr_s = Array(s)
        let arr_t = Array(t)
        
        var ps = 0, pt = 0
        
        while (pt < arr_t.count && ps < arr_s.count) {
            let cs = arr_s[ps]
            let ct = arr_t[pt]
            
            if (cs == ct) {
                ps += 1
                pt += 1
            } else {
                pt += 1
            }
        }
        
        return (ps >= s.count) ? true : false
    }
}

File: code/ex392.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int ns = s.length(), nt = t.length();
        
        vector<vector<bool>> dp(nt + 1, vector<bool>(ns + 1));
        for (int i = 0; i <= nt; i++)    dp[i][0] = 1;
        
        for (int i = 1; i <= nt; i++) {
            for (int j = 1; j <= ns; j++) {
                if (t[i - 1] == s[j - 1]) {
                    dp[i][j] = dp[i - 1][j - 1];
                } else {
                    dp[i][j] = dp[i - 1][j];
                }
            }
        }
        
        return dp[nt][ns];
    }
};